packages/react-components/src/components/VideoGallery/RemoteContentShareReactionOverlay.tsx

```diff
@@ -0,0 +1,244 @@
+/* @conditional-compile-remove(reaction) */
+import { getEmojiResource } from './utils/videoGalleryLayoutUtils';
+/* @conditional-compile-remove(reaction) */
+import {
+  IReactionStyleBucket,
+  generateStartPositionWave,
+  getReactionMovementStyle,
+  getReactionStyleBucket,
+  moveAnimationStyles,
+  opacityAnimationStyles,
+  reactionOverlayStyle,
+  spriteAnimationStyles
+} from '../styles/ReactionOverlay.style';
+/* @conditional-compile-remove(reaction) */
+import {
+  REACTION_NUMBER_OF_ANIMATION_FRAMES,
+  REACTION_SCREEN_SHARE_ANIMATION_TIME_MS,
+  REACTION_START_DISPLAY_SIZE,
+  getCombinedKey,
+  getReceivedUnixTime
+} from './utils/reactionUtils';
+
+/* @conditional-compile-remove(reaction) */
+type VisibleReaction = {
+  reaction: Reaction;
+  id: string;
+  reactionMovementIndex: number;
+  styleBucket: IReactionStyleBucket;
+};
+
+/* @conditional-compile-remove(reaction) */
+type ReceivedReaction = {
+  id: string;
+  status: 'animating' | 'completedAnimating' | 'ignored';
+};
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * The overlay responsible for rendering multiple reactions all at once in presentation mode
+ * @internal
+ */
+export const RemoteContentShareReactionOverlay = React.memo(
+  (props: {
+    reactionResources: ReactionResources;
+    localParticipant?: VideoGalleryLocalParticipant;
+    remoteParticipants?: VideoGalleryRemoteParticipant[];
+    hostDivHeight?: number;
+    hostDivWidth?: number;
+  }) => {
+    const { reactionResources, localParticipant, remoteParticipants, hostDivHeight, hostDivWidth } = props;
+
+    // Reactions that are currently being animated
+    const [visibleReactions, setVisibleReactions] = useState<VisibleReaction[]>([]);
+
+    // Dict of userId to a reaction status. This is used to track the latest received reaction
+    // per user to avoid animating the same reaction multiple times and to limit the number of
+    // active reactions of a certain type.
+    const latestReceivedReaction = useRef<Record<string, ReceivedReaction>>({});
+
+    // Track the number of active reactions of each type to limit the number of active reactions
+    // of a certain type.
+    const activeTypeCount = useRef<Record<string, number>>({
+      like: 0,
+      heart: 0,
+      laugh: 0,
+      applause: 0,
+      surprised: 0
+    });
+
+    // Used to track the total number of reactions ever played. This is a helper variable
+    // to calculate the reaction movement index (i.e. the .left position of the reaction)
+    const visibleReactionPosition = useRef<boolean[]>(new Array<boolean>(55).fill(false));
+
+    const remoteParticipantReactions: Reaction[] = useMemo(
+      () =>
+        remoteParticipants
+          ?.map((remoteParticipant) => remoteParticipant.reaction)
+          .filter((reaction): reaction is Reaction => !!reaction) ?? [],
+      [remoteParticipants]
+    );
+
+    // We can optimize this by applying data structure algorithm and bring this search to log2 N
```

Comment:
Remove comment, this has a max size of 55 so O(n) where n = 55 is totally fine

Reply:
I just liked the problem if the size was massive. Removing the comment.